ics_space_amplification_test.py

```diff
@@ -0,0 +1,192 @@
+            dict_nodes_space_amplification[node.private_ip_address] = node_max_used_capacity_gb - \
+                dict_nodes_initial_capacity[
+                node.private_ip_address]
+            self.log.info(
+                f"Node {node.private_ip_address} used capacity changed from {dict_nodes_initial_capacity[node.private_ip_address]} to {dict_nodes_used_capacity[node.private_ip_address]}.")
+            self.log.info(f"Space amplification is: {dict_nodes_space_amplification[node.private_ip_address]} GB")
+        return dict_nodes_space_amplification
+
+    def _get_nodes_space_amplification_after_write(self, dict_nodes_initial_capacity, written_data_size_gb,
+                                                   start_time) -> dict:
+        self.log.info(f"Space amplification results after a write of: {written_data_size_gb} are:")
+        return self._get_nodes_space_amplification(dict_nodes_initial_capacity=dict_nodes_initial_capacity,
+                                                   start_time=start_time)
+
+    def _get_nodes_used_capacity(self) -> dict:
+        """
+
+        :rtype: dictionary with capacity per node-ip
+        """
+        dict_nodes_used_capacity = {}
+        for node in self.db_cluster.nodes:
+            dict_nodes_used_capacity[node.private_ip_address] = self._get_used_capacity_gb(node=node)
+        return dict_nodes_used_capacity
+
+    @retrying(n=80, sleep_time=60, allowed_exceptions=(AssertionError,))
+    def wait_no_compactions_running(self):
+        compaction_query = "sum(scylla_compaction_manager_compactions{})"
+        now = time.time()
+        results = self.prometheus_db.query(query=compaction_query, start=now - 60, end=now)
+        self.log.debug(f"scylla_compaction_manager_compactions: {results}")
+        # if all are zeros the result will be False, otherwise there are still compactions
+        if results:
+            assert any([float(v[1]) for v in results[0]["values"]]) is False, \
+                "Waiting until all compactions settle down"
+
+    def test_ics_space_amplification(self):  # pylint: disable=too-many-locals
+        # too-many-branches,too-many-statements
+        self._pre_create_schema_with_compaction()
+        stress_queue = list()
+        write_queue = list()
+        verify_queue = list()
+        column_size = 200
+        num_of_columns = 5
+        ops_num = 200200300
+        overwrite_ops_num = ops_num // 2
+        total_new_data_to_write_gb = ops_num * column_size * num_of_columns / (1024 ** 3)
+        total_data_to_overwrite_gb = overwrite_ops_num * column_size * num_of_columns / (1024 ** 3)
+        keyspace_num = 1
+
+        self.log.debug('Test Space-amplification on writing new data')
+        prepare_write_cmd = "cassandra-stress write cl=ALL n={ops_num}  -schema 'replication(factor=3)" \
+                            " compaction(strategy=IncrementalCompactionStrategy)' -port jmx=6868 -mode cql3 native" \
```

Comment:
I thought that cassandra-stress doesn't support this value....

Reply:
it should be fixed by a recent PR of @bhalevy

Reply:
so this PR should also also should have the changes need to use this new c-s version, i.e. it's need a new AMIs with it. @amoskong might be able to help once it's ready.

Reply:
perhaps we better split it to 2 PRs. this is also since PR: https://github.com/scylladb/scylla-enterprise-tools-java/pull/18   is not backported to 2019.1 yet.